### PR TITLE
Avoid missing class error for fetch_thirdparty method

### PR DESCRIPTION
**Pull request by jfefe**

### CODE REVIEW

File: htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -576,6 +576,9 @@ function fetch_thirdparty()
 
         if (empty($this->socid)) return 0;
 
+	if (!class_exists('Societe'))
```
Comment (rdoursenaud) [inline htdocs/core/class/commonobject.class.php]: The "_once" part in "require_once" does exactly that already. The condition is superfluous ;)

**Comment by jfefe** [comment]
You're right, new commit done

**Comment by hregis** [comment]
the class test is better than require_once

**Comment by rdoursenaud** [comment]
@hregis because?

**Comment by hregis** [comment]
@rdoursenaud http://nongnu.13855.n7.nabble.com/require-once-vs-class-exists-td8465.html

**Comment by rdoursenaud** [comment]
Your benchmark is completely irrelevant.

On my production and development machines both methods yield to very similar results (about 20ms for 3000 cycles)

The order of magnitude is so low that it doesn't matter anyway (We're talking about a 0,01ms improvement tops in your test). If you want to improve dolibarr latency (What we're talking about here), you may want to take a look at the translations loading which eats about 50 to 100ms on each request. It's several orders of magnitude more than your tiny little ridicule micro-optimization.

Yes I know I can use memcache to shave that latency, thanks.

Bottom line: require_once is perfectly fine.

**Comment by hregis** [comment]
you have a problem with me, to talk to me like that? :-)
de toute façon dolibarr requière (ou class_exists) une refonte complète ! trop de code en double...

**Comment by rdoursenaud** [comment]
@hregis Nope, not at all. On the contrary, I have great respect for you and the work you've done on Dolibarr.
It's just my way of argue. It's a bit rough, I concur. Sorry if I offended you.

I agree that a great refactoring plan should be established and is sorely needed, but, hey, there's only 24 hours in a day ;)

**Comment by hregis** [comment]
@rdoursenaud I'm sorry too, sometimes badly interprets what is written. Returning to class_exists vs require_once, results of bench, may depend on full settings. (server power, hard drives, etc. ..) but in the end, all optimizations rely butt, can improve speed. regarding translations, would an array instead of parsing a file would not be faster?
